packages/calypso-products/src/plans-utilities.ts

```diff
@@ -41,10 +41,40 @@ export function getTermDuration( term: string ): number | undefined {
 
 export const redirectCheckoutToWpAdmin = (): boolean => !! JETPACK_REDIRECT_CHECKOUT_TO_WPADMIN;
 
-export const is2023PricingGridEnabled = () => {
-	const supportedLocales = [ 'en', 'en-gb', 'es' ];
-	return (
-		isEnabled( 'onboarding/2023-pricing-grid' ) &&
-		supportedLocales.includes( getLocaleSlug() ?? '' )
-	);
+/**
+ * Returns if the 2023 Pricing grid optimized design should be visible or not
+ *
+ */
+export const is2023PricingGridEnabled = (): boolean => {
+	const isFeatureFlagEnabled = (): boolean => isEnabled( 'onboarding/2023-pricing-grid' );
+
+	const isLocaleSupported = (): boolean => {
+		const supportedLocales = [ 'en', 'en-gb', 'es' ];
+		return supportedLocales.includes( getLocaleSlug() ?? '' );
+	};
+
+	const isActivePage = (): boolean => {
+		let currentRoutePath = '';
+		if ( typeof window === 'object' && window.location ) {
+			currentRoutePath = window.location.pathname;
+		}
+
+		const [ firstPathPart, secondPathPart ] = ( currentRoutePath || '' )
+			.split( '/' )
+			.filter( ( i ) => i );
+
+		// Is this the internal plans page /plans/<site-slug> ?
+		if ( firstPathPart === 'plans' ) {
+			return true;
+		}
+
+		// Is this the onboarding flow /start/plans?
+		if ( firstPathPart === 'start' && secondPathPart === 'plans' ) {
+			return true;
+		}
+
+		return false;
```

Comment:
This is unnecessarily complex for what you are describing. What is the point of splitting the string even? 

If regex seems scary, then I'd opt for some native method that does the string matching e.g. 

> Is this the internal plans page /plans/<site-slug> ?

`pathname.startsWith( '/plans/' )`

> Is this the onboarding flow /start/plans?

`pathname.startsWith( '/start/plans/' )`

Reply:
> If regex seems scary

for regex to fully capture both cases you mentioned, this looks like the pattern: `/^\/(plans|start\/plans)(\/.*)?$/gm`. It's simple, just both patterns congested together. The `(\/.*)?` portion is the optional second part (trailing slash and everything after). These look like the cases you capture with the manual string splitting.

`/plans` match
`/plans/` match
`/plans/foo` match
`/plansfoo` ~no match~
`plansfoo` ~no match~

`/start/plans` match
`/start/plans/` match
`/start/plans/foo` match
`/start/plansfoo` ~no match~
`start/plansfoo` ~no match~

`/` ~no match~
`/foo` ~no match~
`/foo/bar` ~no match~

https://regex101.com/r/fgvoLz/2

Reply:
> If regex seems scary

Haha well not scary, but, There is just this nagging feeling I might have missed something when I put into regex!

Need to up my regex game 😎 

Thanks for the exact pattern Christos.

Reply:
> This is unnecessarily complex for what you are describing. What is the point of splitting the string even?

Wouldn't this complexity be warranted because this will be used frequently? When comparing to regex wouldn't a string split and a comparison be more performant, given that we know the exact use cases that we would be using? 

👍 for using the native method!

> pathname.startsWith( '/start/plans' )

Reply:
I think the bottom line is it shouldn't be used frequently. Not more than once per path change, so I don't know.